recipes/gtk/all/patches/0001-fix-windows-rc-error.patch

```diff
@@ -0,0 +1,25 @@
+winnt.h requires architecture defines such as _AMD64_. This is defined in Windows.h but not in winuser.h
+diff --git a/tools/generate-uac-manifest.py b/tools/generate-uac-manifest.py
+index ede6ba5..783a6e4 100644
+--- a/tools/generate-uac-manifest.py
++++ b/tools/generate-uac-manifest.py
+@@ -92,7 +92,7 @@ def write_rc_file(name, resource, outdir):
+ 
+     if resource is None:
+         outfile = open(output_file_base_name + '.rc', 'w+')
+-        outfile.write('#include <winuser.h>')
++        outfile.write('#include <Windows.h>')
+     else:
+         if resource != output_file_base_name + '.rc':
+             outfile = open(output_file_base_name + '.rc', 'w+')
+diff --git a/gtk/gtk-win32.rc.body.in b/gtk/gtk-win32.rc.body.in
+index b876a6d..ff16e7c 100644
+--- a/gtk/gtk-win32.rc.body.in
++++ b/gtk/gtk-win32.rc.body.in
+@@ -1,5 +1,4 @@
+-#include <winuser.h>
+-#include <winver.h>
++#include <Windows.h>
+ 
+ GTK_ICON		ICON			"gtk.ico"
+ 
```

Comment:
So you managed to find the fix, great! 
Does it come from upstream ? did you submit it ?

Reply:
No it does not come from upstream and I did not submit it yet. I think this is related to the version of Windows SDK, and I'm not very sure whether it is the correct solution. Will discuss this with upstream maintainers later.

I checked the log and find this line and noticed that I recently went into the same problem:
```
C:\Program Files (x86)\Windows Kits\10\include\10.0.22000.0\um\winnt.h(253) : error RC2188
```

Reply:
I checked and they have already fixed this. Let me use their patch instead.